src/refract/RenderJSONVisitor.cc

```diff
@@ -0,0 +1,153 @@
+//
+//  refract/RenderJSONVisitor.cc
+//  librefract
+//
+//  Created by Pavan Kumar Sunkara on 17/06/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#include "Element.h"
+#include "Visitors.h"
+#include "sosJSON.h"
+#include <sstream>
+
+namespace refract
+{
+
+    RenderJSONVisitor::RenderJSONVisitor(const sos::Base::Type& type_)
+    : type(type_), isExtend(false) {}
+
+    RenderJSONVisitor::RenderJSONVisitor()
+    : type(sos::Base::UndefinedType), isExtend(false) {}
+
+    void RenderJSONVisitor::assign(sos::Base value) {
+        if (isExtend) {
+            extend(value);
+        }
+        else if (type == sos::Base::ArrayType) {
+            pArr.push(value);
+        }
+        else if (type == sos::Base::UndefinedType) {
+            result = value;
+        }
+    }
+
+    void RenderJSONVisitor::assign(std::string key, sos::Base value) {
+        if (!key.empty() && type == sos::Base::ObjectType) {
+            pObj.set(key, value);
+        }
+    }
+
+    void RenderJSONVisitor::extend(sos::Base value) {
+        if (type == sos::Base::ObjectType) {
+
+            for (sos::KeyValues::iterator it = value.object().begin();
+                 it != value.object().end();
+                 ++it) {
+
+                pObj.set(it->first, it->second);
+            }
+        }
+        else if (type == sos::Base::ArrayType) {
+
+            for (sos::Bases::iterator it = value.array().begin();
+                 it != value.array().end();
+                 ++it) {
+
+                pArr.push(*it);
+            }
+        }
+    }
+
+    void RenderJSONVisitor::visit(const IElement& e) {
+        e.content(*this);
+    }
+
+    void RenderJSONVisitor::visit(const MemberElement& e) {
+        RenderJSONVisitor renderer;
+        renderer.visit(*e.value.second);
+        assign(static_cast<const StringElement&>(*e.value.first).value, renderer.get());
```

Comment:
I would like to merge the Reference expansion into this branch and once tests pass, we will merge this branch.